app/src/test/java/org/mozilla/fenix/IntentReceiverActivityTest.kt

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix
+
+import android.content.Intent
+import android.speech.RecognizerIntent.ACTION_RECOGNIZE_SPEECH
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import mozilla.components.support.test.argumentCaptor
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.eq
+import org.mockito.Mockito.never
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.`when`
+import org.robolectric.annotation.Config
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.settings
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+
+@ObsoleteCoroutinesApi
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class IntentReceiverActivityTest {
+
+    @Test
+    fun `process intent with alwaysOpenInPrivateMode set to true`() {
+        runBlockingTest {
+            testContext.settings.alwaysOpenInPrivateMode = true
+
+            val intent = Intent()
+            `when`(testContext.components.intentProcessors.privateIntentProcessor.process(intent)).thenReturn(true)
+            val activity = Robolectric.buildActivity(IntentReceiverActivity::class.java, intent).get()
+            activity.processIntent(intent)
+
+            // Not using mockk here because process is as suspend function
```

Comment:
nit typo on all of these comments: "is as suspend function" -> "is a suspend function"

Reply:
Ah, thanks for fixing that!